### PR TITLE
Feature/output change as ringbuffer

### PR DESCRIPTION
**Pull request by jmvdwerf**
Rewrote the linked list as a ringbuffer.

### CODE REVIEW

File: src/domotica/domotica.c [hunk 1]
```diff
@@ -22,48 +22,54 @@ __attribute__ ((weak, alias ("domotica_handle_output_change_dummy"))) \
...
+
+static OUTPUT_CHANGE_RINGBUFFER_Type buffer = { { { 0, 0} }, 0, 0};
 
 // ------------------------------------------------------------------
 void domotica_enqueue_output_change(uint16_t mask_on, uint16_t mask_off)
 {
-  OUTPUT_CHANGE_Type *change = malloc(sizeof(OUTPUT_CHANGE_Type));
-  change->mask_on = mask_on;
-  change->mask_off = mask_off;
+  (void) mask_on;
```
Comment (fuegas) [inline src/domotica/domotica.c]: Deze voids mogen weg omdat je ze gebruikt in je functie.

**Review by fuegas — CHANGES_REQUESTED** [review]

File: src/domotica/domotica.c (hunk 1 above)
Comment (jmvdwerf) [reply src/domotica/domotica.c]: Changed!

**Comment by jmvdwerf** [comment]
Consider it done!

File: src/domotica/domotica.c [hunk 2]
```diff
@@ -22,48 +22,52 @@ __attribute__ ((weak, alias ("domotica_handle_output_change_dummy"))) \
...
   uint16_t mask_off;
-  struct OUTPUT_CHANGE *next;
 } OUTPUT_CHANGE_Type;
 
-static OUTPUT_CHANGE_Type *output_change_queue = 0;
+typedef struct {
+  OUTPUT_CHANGE_Type buffer[DOMOTICA_CHANGE_BUFFER_Size];
+  volatile uint8_t writer;
+  volatile uint8_t reader;
+} OUTPUT_CHANGE_RINGBUFFER_Type;
```
Comment (fuegas) [inline src/domotica/domotica.c]: `OUTPUT_CHANGE_Type `, `DOMOTICA_CHANGE_BUFFER_Size` and `OUTPUT_CHANGE_RINGBUFFER_Type` are not consistent in naming.

**Review by fuegas — CHANGES_REQUESTED** [review]

File: src/domotica/domotica.c (hunk 2 above)
Comment (jmvdwerf) [reply src/domotica/domotica.c]: Big reason is that the OUTPUT_CHANGE_ are internal variables that represent the buffer (and are static with a comprehensive name), whereas DOMOTICA_CHANGE_BUFFER_Size can be externally defined. For naming I found this more intuitive than having DOMOTICA_OUTPUT_CHANGE_... as a name...

**Comment by jmvdwerf** [comment]
Changed!

File: src/domotica/domotica.c (hunk 2 above)
Comment (jmvdwerf) [reply src/domotica/domotica.c]: I changed the names, as it only appears in the top part. Also the OUTPUT_CHANGE_Type needs to be exported for the next release to add priorities to it...

File: screen_layout [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# the following two lines give a two-line status, with the current window highlighted
```
Comment (fuegas) [inline screen_layout]: Please leave editor preferences out of the repository.

You can use them and keep them untracked by adding them to the global gitignore or the local ignore in `.git/info/exclude`.

**Review by fuegas — CHANGES_REQUESTED** [review]

File: screen_layout (hunk 3 above)
Comment (jmvdwerf) [reply screen_layout]: Deze zou eigenlijk in het hoofdproject geinclude moeten worden, maar dat is mijn persoonlijke mening... Dit is geen editor, maar screen...

**Review by fuegas — APPROVED** [review]

**Pull request merged**